tests/integration/tests.mk

```diff
@@ -105,6 +105,21 @@ test.integration.kube: | $(JUNIT_REPORT)
 	${_INTEGRATION_TEST_WORK_DIR_FLAG} \
 	2>&1 | tee >($(JUNIT_REPORT) > $(JUNIT_UNIT_TEST_XML))
 
+# Integration tests that detect race condition for Kubernetes environment.
+.PHONY: test.integration.race.kube
+test.integration.race.kube: | $(JUNIT_REPORT)
+	mkdir -p $(dir $(JUNIT_UNIT_TEST_XML))
+	set -o pipefail; \
+	$(GO) test -race -p 1 ${T} ${TEST_PACKAGES} ${_INTEGRATION_TEST_WORKDIR_FLAG} ${_INTEGRATION_TEST_CIMODE_FLAG} -timeout 120m \
```

Comment:
This doesn't do anything, the binaries run in kube without the -race tag applied. You would need to build the images with `-race`. This just detects race conditions in the test framework which doesn't really matter.

This IS useful for native though

Reply:
Added the tests (test.integration.race.native) for the native environment. The tests (test.integration.race.kube) currently only detect the race condition in the integraion test directory (tests/integration); enabling race detection in all Istio images will be a separate PR.

Reply:
I don't think we should turn on the kube ones right now until they are actually testing something

Reply:
The kube ones right now only detect the race condition for the code in the integraion test directory (tests/integration), which is the code of test framework. Ideally, besides the test framework, the kube ones should also detect race condition for all Istio images (Istio images are currently not built with -race flag due to CPU and memory overhead). Since the kube ones detect the race condition in the test framework, they are still of value and they are only placed in the post-submit tests.

Reply:
We already test the test framework with the local one. Without the images using race detection as well I think this is a waste (in terms of test time, test resources, and dev time).

Reply:
Done: removed the kube ones.